tools/ocirepo/ocirepo.go

```diff
@@ -26,14 +26,16 @@ func (o *OCIRepo) GetComponentVersion(archive *comparch.ComponentArchive,
 	return version, nil
 }
 
-func (o *OCIRepo) PushComponentVersionIfNotExist(archive *comparch.ComponentArchive, repo cpi.Repository) error {
+func (o *OCIRepo) PushComponentVersion(archive *comparch.ComponentArchive, repo cpi.Repository,
+	overwrite bool,
+) error {
 	if exists, _ := repo.ExistsComponentVersion(archive.GetName(),
-		archive.GetVersion()); exists {
+		archive.GetVersion()); exists && !overwrite {
 		return fmt.Errorf("cannot push component version %s: %w",
 			archive.GetVersion(), errComponentVersionAlreadyExists)
 	}
```

Comment:
This is completely personal preference, but I typically assume that a `if <declaration>; <predicate>` block has a predicate that only checks things coming out of the declaration. It is just easy to oversee that there is also external state involved. So here I would prefer something like below. But this is likely only me, pls feel free to ignore this comment in case you don't see it as an issue.

```go
exists, _ := repo.ExistsComponentVersion(archive.GetName(), archive.GetVersion())
if exists && !overwrite {
	return fmt.Errorf("cannot push component version %s: %w",
		archive.GetVersion(), errComponentVersionAlreadyExists)
}
```